injected/src/features/broker-protection/actions/expectation.js

```diff
@@ -1,23 +1,43 @@
 import { getElement } from '../utils.js'
 import { ErrorResponse, SuccessResponse } from '../types.js'
+import { execute } from '../execute.js'
 
 /**
  * @param {Record<string, any>} action
- * @param {Document | HTMLElement} root
+ * @param {Record<string, any>} userData
+ * @param {Document} root
  * @return {import('../types.js').ActionResponse}
  */
-export function expectation (action, root = document) {
+export function expectation (action, userData, root = document) {
+    /**
+     * @type {Array<import('../types.js').BooleanResult & { failSilently?: boolean }>}
+     */
     const results = expectMany(action.expectations, root)
+    const errors = []
+    let runActions = true
+
+    results.forEach((x) => {
+        if (x.result === false) {
+            runActions = false
+
+            if (!x.failSilently) {
+                errors.push('error' in x ? x.error : 'unknown error')
+            }
 
-    const errors = results.filter(x => x.result === false).map(x => {
-        if ('error' in x) return x.error
-        return 'unknown error'
+            delete x.failSilently
```

Comment:
we can avoid passing this around, see https://github.com/duckduckgo/content-scope-scripts/commit/3676158835a8325108db37e81933d0fff462b1fa for a possible approach - the idea is just to reach back to the original array using the index. It will prevent needing to change anything else in this file